github/pkg/reconciler/source/githubsource.go

```diff
@@ -248,6 +275,42 @@ func (r *Reconciler) reconcileReceiveAdapter(ctx context.Context, source *source
 	return ksvc, err
 }
 
+func (r *Reconciler) reconcileKIngress(ctx context.Context, source *sourcesv1alpha1.GitHubSource) (*networkingv1alpha1.Ingress, error) {
+	host, err := domains.DomainNameFromTemplate(ctx, source.ObjectMeta, source.Name)
```

Comment:
This is really not intended for public consumption.

Reply:
Yes. KIngress itself is not intended for public consumption (right?). For this to work we need a Knative Serving to support this feature as part of its public API.

Reply:
Consuming kingress elsewhere in Knative is on the table.  Reaching inside of the way the Route controller drives kingress is a very different thing.